internal/api/mocks.go

```diff
@@ -405,22 +405,17 @@ func DefaultQuerier(contractAddr string, coins types.Coins) types.Querier {
 }
 
 func (q MockQuerier) Query(request types.QueryRequest, _gasLimit uint64) ([]byte, error) {
-	marshaled, err := json.Marshal(request)
-	if err != nil {
-		return nil, err
-	}
-	q.usedGas += uint64(len(marshaled))
```

Comment:
Why this deletion? Tere idea is to have an input specific gas usage for mocking.

Reply:
go tells me it is unused code-- think it is used?

Reply:
This one is tricky. golangci-lint complains with the following:

```
internal/api/mocks.go:412:2: SA4005: ineffective assignment to field MockQuerier.usedGas (staticcheck)
        q.usedGas += uint64(len(marshaled))
        ^
```

where SA4005 is here: https://staticcheck.io/docs/checks#SA4005.

So it seems like we should turn

```
- func (q MockQuerier) Query(request types.QueryRequest, _gasLimit uint64) ([]byte, error) {
+ func (q *MockQuerier) Query(request types.QueryRequest, _gasLimit uint64) ([]byte, error) {
```

but that also creates all sorts of other problems. And wasmd uses `q` by value and not by pointer too.

@alpe do you have any idea how to resolve this nicely?

@faddat if you can revert those lines, we can merge this PR.

Reply:
The MockQuerier should be a pointer, indeed. Otherwise the `usedGas` would not be updated. The changes are minimal: See my `427_pointers` branch.

A test would have detected that `usedGas` is not updated. So IMHO it is ok to just drop it here. It is an internal test type and can be modified when needed.

Reply:
Thanks, Alex. I applied the change [here](https://github.com/CosmWasm/wasmvm/pull/429) because your commit contains some unrelated thing.